ballerina/client.bal

```diff
@@ -357,760 +358,425 @@ public isolated client class Client {
-        json|http:ClientError response = self.salesforceClient->get(path);
-        if response is json {
-            return response;
+    // Added APIs
+    
+    # Retrieves the list of individual records that have been deleted within the given timespan.
+    #
+    # + sObjectName - SObject reference
+    # + startDate - Start date of the timespan.
+    # + endDate - End date of the timespan.
+    # + return - `DeletedRecordsResult` record if successful or else `error`
+    isolated remote function getDeleted(string sObjectName, time:Civil startDate, time:Civil endDate) 
+        returns DeletedRecordsResult|error {
+        string path = utils:prepareUrl([API_BASE_PATH, SOBJECTS, sObjectName, DELETED]);
+        string finalUrl = utils:addQueryParameters(path,{'start: check time:civilToString(removeDecimalPlaces(startDate)), 
+            end: check time:civilToString(removeDecimalPlaces(endDate))});
+        return check self.salesforceClient->get(finalUrl);
+    }
+
+    # Retrieves the list of individual records that have been deleted within the given timespan.
+    #
+    # + sObjectName - SObject reference
+    # + startDate - Start date of the timespan.
+    # + endDate - End date of the timespan.
+    # + return - `UpdatedRecordsResults` record if successful or else `error`
+    isolated remote function getUpdated(string sObjectName, time:Civil startDate, time:Civil endDate) 
+        returns UpdatedRecordsResults|error {
+        string path = utils:prepareUrl([API_BASE_PATH, SOBJECTS, sObjectName, UPDATED]);
+        string finalUrl = utils:addQueryParameters(path,{'start: check time:civilToString(removeDecimalPlaces(startDate)), 
+            end: check time:civilToString(removeDecimalPlaces(endDate))});
+        return check self.salesforceClient->get(finalUrl);
+    }
+
+    # Get the password information
+    #
+    # + userId - User ID
+    # + return - `boolean` if successful or else `error`
+    isolated remote function getPasswordInfo(string userId) returns boolean|error {
+        string path = utils:prepareUrl([API_BASE_PATH, SOBJECTS, USER, userId, PASSWORD]);
+        http:Response response = check self.salesforceClient->get(path);
+        if response.statusCode == 200 {
+            json payload = check response.getJsonPayload();
+            map<json> payloadMap = check payload.ensureType();
+            boolean isExpired = check payloadMap.isExpired;
+            return isExpired;
         } else {
-            return checkAndSetErrorDetail(response);
+            json payload = check response.getJsonPayload();
+            json[] jsonArray= check payload.ensureType();
+            map<json> payloadMap = check jsonArray[0].ensureType();
+            record {string message; string errorCode;} responseRecord = {message: check payloadMap.message, errorCode: check payloadMap.errorCode};
+            return error("Error occurred while resetting the password" + responseRecord.toString());
```

Comment:
is the error message correct?
instead of appending, shall we pass it as named arguments
```
error err = error("Whoops", httpCode = 27);
```

Reply:
Check all places

Reply:
Updated in [update code according to comments](https://github.com/ballerina-platform/module-ballerinax-sfdc/pull/311/commits/cd6ae536defa51bd247b84742f1c480c2cf80ab1)